src/gui/callstatechecker.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2022 by Claudio Cambra <claudio.cambra@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#pragma once
+
+#include <QObject>
+#include <QTimer>
+
+#include "networkjobs.h"
+#include "accountstate.h"
+
+namespace OCC {
+
+class CallStateChecker : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(QString token READ token WRITE setToken NOTIFY tokenChanged)
+    Q_PROPERTY(AccountState* accountState READ accountState WRITE setAccountState NOTIFY accountStateChanged)
+
+public:
+    explicit CallStateChecker(QObject *parent = nullptr);
+
+    QString token() const;
+    void setToken(const QString &token);
+
+    AccountState* accountState() const;
+    void setAccountState(AccountState *accountState);
+
+signals:
+    void tokenChanged();
+    void accountStateChanged();
+
+    void callJoined();
+    void callAbandoned();
+    void notificationTimerElapsed();
+
+public slots:
+    void startChecking();
+    void stopChecking();
```

Comment:
should we not better use a property named `checking` that would be set to `true` when we want the checking to occur ?
sounds easier to then set a binding on the qml side

Reply:
I replaced `startChecking` and `stopChecking` with a property